lib/galaxy/dependencies/conditional_requirements.txt

```diff
@@ -0,0 +1,10 @@
+# These dependencies are only required when certain config options are set
+psycopg2==2.6.1
+WebError==0.11
+python-openid
+MySQL-python
+fluent-logger
+raven
+pbs_python
+drmaa
+# PyRods not in PyPI
```

Comment:
- `boto`: Added the conditional for this only to discover that it's already a dependency of non-optional `bioblend`, so I added it to `requirements.txt` instead.
- `poster`: Is not in Pulsar's `requirements.txt` or `setup.py` yet seems to be required if not using `pycurl` or `requests` transport. @jmchilton ?
- `requests`: Pulsar will gracefully fall back to `poster`+`urllib/urllib2` (use of `requests` is not explicitly configurable).
- `python_daemon`: I think it's pretty unlikely anyone is actually using the transfer manager, which is the only thing that uses python_daemon.
- `twill`: Automatically installed into the tox venvs by tox, not needed for Galaxy otherwise.